ui/src/pages/custody/Accounts.tsx

```diff
@@ -42,10 +45,42 @@ const AccountsComponent: React.FC<RouteComponentProps & Props> = ({
 
   const accounts = useStreamQueries(AssetSettlementRule).contracts;
   const assets = useStreamQueries(AssetDescription).contracts;
+  const deposits = useStreamQueries(AssetDeposit).contracts;
 
   const clientServices = services.filter(s => s.payload.customer === party);
   const assetNames = assets.map(a => a.payload.description);
 
+  const addSignatoryAsDepositObserver = async (
+    deposit: CreateEvent<AssetDeposit>,
+    newObs: string[]
+  ) => {
+    const newObservers = makeDamlSet([...deposit.observers, ...newObs]);
+
+    await ledger.exercise(AssetDeposit.AssetDeposit_SetObservers, deposit.contractId, {
+      newObservers,
+    });
+  };
+
+  const updateDeposits: any = async (retries: number) => {
+    if (retries > 0) {
+      return updateDeposits(retries - 1);
+    }
+    return Promise.all(
+      deposits.map(d => {
+        const newObservers = damlSetValues(d.payload.asset.id.signatories).filter(
+          signatory => !d.observers.includes(signatory)
+        );
+        if (newObservers.length > 0) {
+          addSignatoryAsDepositObserver(d, newObservers);
+        }
+      })
+    );
+  };
+
+  useEffect(() => {
+    updateDeposits(3);
+  }, []);
```

Comment:
I think we'll have to come up with another solution for this problem as this solution will cause issues for any services that have a stored `AssetDeposit` contract id on their contract (Ie, Trading, Custody, Distribution, etc.).  


Looking at Eric's messages on Slack
> it's much better if the issuer has their own cap table and holds all shares themselves. and can even have customers hold assets on the issuer's cap table.
 
I'm thinking we can solve this by having the issuer be a custody provider and clients of the issuer can open accounts at said issuer. This would mean the cap table and the shares would both be owned by the issuer and solve Erics problem. What do you think ?

Reply:
yes I removed this from the pr, should be updated as of my latest commit

Reply:
Because it deserves its own pr, definitely

Reply:
I agree I think this needs to be redone, and I like that approach !